src/Test/Utilities/CLRHelpers.cs

```diff
@@ -64,12 +65,14 @@ public static string[] PeVerify(string filePath)
             return PeVerify(File.ReadAllBytes(filePath), AppDomain.CurrentDomain.Id, filePath);
         }
 
-        private static readonly object Guard = new object();
+        private static readonly Mutex s_guard = new Mutex(false, new Guid().ToString());
```

Comment:
Is `s_guard` shared between AppDomains, or does each AppDomain get a different copy? If it's the latter then how does that help?